### PR TITLE
[Java] Remove non-raylet code in Java.

### PR DESCRIPTION
**Pull request by jovany-wang**
## What do these changes do?
Because there are no new features in non-raylet mode in Java. We needn't to provide the non-raylet option in Java now.

The changes of this PR:
1) Removing `use_raylet` flag and changing the code that used the flag.

2) Removing some parameters that was used in non-raylet mode, like `local_scheduler`, `object_manager` and `global_scheduler` and changeing the related code.

3) Removing class `BaseStateStoreProxyImpl` and `NonRayletStateStoreProxyImpl.java`.

4) Fix a small bug and improve some code's readability.

## Related issue number
N/A

### CODE REVIEW

File: java/runtime-common/src/main/java/org/ray/spi/LocalSchedulerLink.java [hunk 1]
```diff
@@ -13,17 +16,13 @@
...
-  void reconstructObject(UniqueId objectId, boolean fetchOnly);
-
   void reconstructObjects(List<UniqueId> objectIds, boolean fetchOnly);
 
   void notifyUnblocked();
 
   UniqueId generateTaskId(UniqueId driverId, UniqueId parentTaskId, int taskIndex);
 
-  List<byte[]> wait(byte[][] objectIds, int timeoutMs, int numReturns);
+  <T> WaitResult<T> wait(List<RayObject<T>> waitFor, int numReturns, int timeout);
```
Comment (raulchen) [inline java/runtime-common/src/main/java/org/ray/spi/LocalSchedulerLink.java]: `timeoutMs`

File: java/runtime-dev/src/main/java/org/ray/spi/impl/MockLocalScheduler.java [hunk 2]
```diff
@@ -96,8 +89,8 @@ public UniqueId generateTaskId(UniqueId driverId, UniqueId parentTaskId, int tas
   }
 
   @Override
-  public List<byte[]> wait(byte[][] objectIds, int timeoutMs, int numReturns) {
-    return store.wait(objectIds, timeoutMs, numReturns);
+  public <T> WaitResult<T> wait(List<RayObject<T>> waitFor, int numReturns, int timeout) {
+    throw new RuntimeException("Not implemented here.");
```
Comment (raulchen) [inline java/runtime-dev/src/main/java/org/ray/spi/impl/MockLocalScheduler.java]: It shouldn't throw exception here. But I think `MockLocalScheduler` is already broken, let's fix it later.

File: java/runtime-native/src/main/java/org/ray/runner/RunInfo.java [hunk 3]
```diff
@@ -33,9 +33,7 @@
     return processes;
   }
 
-  public enum ProcessType {
-    PT_WORKER, PT_LOCAL_SCHEDULER, PT_PLASMA_MANAGER, PT_PLASMA_STORE,
-    PT_GLOBAL_SCHEDULER, PT_REDIS_SERVER, PT_WEB_UI, PT_RAYLET,
-    PT_DRIVER
+  public enum ProcessType { PT_WORKER, PT_PLASMA_STORE,
+    PT_REDIS_SERVER, PT_WEB_UI, PT_RAYLET, PT_DRIVER
```
Comment (raulchen) [inline java/runtime-native/src/main/java/org/ray/runner/RunInfo.java]: put one item each line.

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java [hunk 4]
```diff
@@ -28,78 +30,60 @@
...
     client = nativeInit(schedulerSockName, clientId.getBytes(),
-        isWorker, driverId.getBytes(), useRaylet);
-    this.useRaylet = useRaylet;
+        isWorker, driverId.getBytes());
   }
 
   @Override
-  public List<byte[]> wait(byte[][] objectIds, int timeoutMs, int numReturns) {
-    assert (useRaylet == true);
+  public <T> WaitResult<T> wait(List<RayObject<T>> waitFor, int numReturns, int timeoutMillis) {
```
Comment (raulchen) [inline java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: `timeoutMillis` -> `timeoutMs`, and also change the super class

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java [hunk 5]
```diff
@@ -28,78 +30,60 @@
...
+    List<UniqueId> ids = new ArrayList<>();
+    for (RayObject<T> element : waitFor) {
+      ids.add(element.getId());
+    }
+
+    boolean[] readys = nativeWaitObject(client, getIdBytes(ids), numReturns, timeoutMillis, false);
```
Comment (raulchen) [inline java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: `readys` -> `ready`

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java [hunk 6]
```diff
@@ -28,78 +30,60 @@
...
+    if (readys.length != ids.size()) {
```
Comment (raulchen) [inline java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: why would this happen?

File: java/runtime-common/src/main/java/org/ray/spi/ObjectStoreProxy.java [hunk 7]
```diff
@@ -89,39 +82,6 @@ public void put(UniqueId id, Object obj, Object metadata) {
...
-  public <T> WaitResult<T> wait(List<RayObject<T>> waitfor, int numReturns, int timeout) {
-    List<UniqueId> ids = new ArrayList<>();
-    for (RayObject<T> obj : waitfor) {
-      ids.add(obj.getId());
-    }
-    List<byte[]> readys;
-    if (localSchedulerLink == null) {
-      readys = store.wait(getIdBytes(ids), timeout, numReturns);
-    } else {
-      readys = localSchedulerLink.wait(getIdBytes(ids), timeout, numReturns);
```
Comment (raulchen) [inline java/runtime-common/src/main/java/org/ray/spi/ObjectStoreProxy.java]: `getIdBytes` can also be removed?

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java (hunk 6 above)
Comment (jovan-wong) [reply java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: em, this check seems to be unnecessary.

File: java/runtime-common/src/main/java/org/ray/spi/ObjectStoreProxy.java (hunk 7 above)
Comment (jovan-wong) [reply java/runtime-common/src/main/java/org/ray/spi/ObjectStoreProxy.java]: It couldn't be removed because of some references by other methods.

**Comment by jovan-wong** [comment]
@AmplabJenkins  retest this please.

**Comment by jovan-wong** [comment]
@AmplabJenkins  Retest this please.

**Comment by jovan-wong** [comment]
@AmplabJenkins  retest this, please.

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java (hunk 6 above)
Comment (raulchen) [reply java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: yep, let's remove it then.

File: java/runtime-common/src/main/java/org/ray/spi/ObjectStoreProxy.java (hunk 7 above)
Comment (raulchen) [reply java/runtime-common/src/main/java/org/ray/spi/ObjectStoreProxy.java]: ok, I thought it wasn't used any more.

File: java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java (hunk 6 above)
Comment (jovan-wong) [reply java/runtime-native/src/main/java/org/ray/spi/impl/DefaultLocalSchedulerClient.java]: removed.

**Comment by guoyuhong** [comment]
@AmplabJenkins retest this, please.

**Comment by jovan-wong** [comment]
@raulchen  @robertnishihara cc again?

**Review by robertnishihara — APPROVED** [review]

**Comment by robertnishihara** [comment]
Thanks @jovan-wong this is great! It looks good to me.

**Pull request merged**